test/unit/master/test_build.py

```diff
@@ -273,6 +273,25 @@ def test_cancel_is_a_noop_if_build_is_already_finished(self):
         self.assertEqual(len(mock_slave.method_calls), num_slave_calls_before_cancel,
                          'Canceling a finished build should not cause any further calls to slave.')
 
+    def test_cancel_calls_kill_subprocesses_to_set_kill_event(self):
+        mock_slave = self._create_mock_slave()
+        build = self._create_test_build(BuildStatus.QUEUED, slaves=[mock_slave])
+        self.patch('app.master.build.util')
+        build.generate_project_type()
+
+        build.cancel()
+
+        build._project_type.kill_subprocesses.assert_called_once_with()
```

Comment:
This will fail if we ever make `kill_subprocess` accept arguments.

How about
```
self.assertTrue(build._project_type.kill_subprocess.called,
                          'Build did not attempt to kill subprocess')
```

Also, why is the build trying to kill a subprocess anyway if it's in the QUEUED state? It shouldn't have a subprocess unless it's PREPARING, correct?